packages/studio-base/src/panels/ThreeDeeRender/renderables/Images/WorkerImageDecoder.ts

```diff
@@ -39,12 +39,19 @@ export class WorkerImageDecoder {
     options: Partial<RawImageOptions>,
   ): Promise<ImageData> {
     return await new Promise((resolve, reject) => {
+      const oldAbort = this.#abort;
       // abort previous request
-      if (this.#abort) {
-        this.#abort();
-      }
-      this.#abort = reject;
-      void this.#remote.decode(image, options).then(resolve).catch(reject);
+      this.#abort = () => {
+        reject("Decode aborted.");
```

Comment:
We should reject using error instances rather than free strings so we have stack traces.

```suggestion
        reject(new Error("Decode aborted."));
```